pandas/_libs/tslib.pyx

```diff
@@ -457,26 +461,35 @@ cpdef array_to_datetime(
                 elif PyDateTime_Check(val):
                     seen_datetime = True
                     if val.tzinfo is not None:
-                        if utc_convert:
-                            _ts = convert_datetime_to_tsobject(val, None)
-                            iresult[i] = _ts.value
-                        else:
+                        seen_aware = True
+
+                        if seen_tz is None:
```

Comment:
What happens when we get a timezone offset from a string and a tzinfo object from a `datetime.datetime`?